agents/template/src/main/scala/org/broadinstitute/transporter/kafka/TransferStreamBuilder.scala

```diff
@@ -127,42 +123,32 @@ class TransferStreamBuilder[I: Decoder, P: Encoder: Decoder, O: Encoder](
 
             val (stepsSoFar, message) = progress.message
 
-            message
-              .as[P]
-              .flatMap(parsed => Either.catchNonFatal(runner.step(parsed)))
-              .fold(
-                err => {
-                  val message =
-                    s"Failed to run step on transfer ${ids.transfer} from request ${ids.request}"
-                  logger.error(err)(message)
-                  val errSummary = TransferMessage(
-                    ids,
-                    (
-                      TransferResult.FatalFailure: TransferResult,
-                      UnhandledErrorInfo(message, Option(err.getMessage))
-                    ).asJson
-                  )
-                  Either.right(errSummary)
-                },
-                nextStepOrOutput => {
-                  logger.info(
-                    s"Successfully ran transfer step for ${ids.transfer} from request ${ids.request}"
-                  )
-                  nextStepOrOutput.bimap(
-                    nextStep => TransferMessage(ids, (stepsSoFar + 1) -> nextStep.asJson),
-                    output =>
-                      TransferMessage(
-                        ids,
-                        (TransferResult.Success: TransferResult, output).asJson
-                      )
-                  )
-                }
-              )
+            (
+              ids,
+              stepsSoFar + 1,
+              message
+                .as[P]
+                .flatMap(parsed => Either.catchNonFatal(runner.step(parsed)))
+                .fold(
+                  err => Failure(err),
+                  identity
+                )
```

Comment:
I'd pull the chain from `message.as[P]` to here out into a local `val`, for better readability